docs/security_self_assessment.md

```diff
@@ -0,0 +1,1588 @@
+
+Remediation: RKE2 doesn’t require or maintain a configuration file for the controller manager. All configuration is passed in as arguments at container run time.
+
+*1.1.5* Ensure that the scheduler pod specification file permissions are set to `644` or more restrictive (Scored)
+
+Result: Not Applicable
+
+Remediation: RKE2 doesn’t require or maintain a configuration file for the scheduler. All configuration is passed in as arguments at container run time.
+
+*1.1.6* Ensure that the scheduler pod specification file ownership is set to `root:root` (Scored)
+
+Result: Not Applicable
+
+Remediation: RKE2 doesn’t require or maintain a configuration file for the scheduler. All configuration is passed in as arguments at container run time.
+
+*1.1.7* Ensure that the etcd pod specification file permissions are set to `644` or more restrictive (Scored)
+
+Result: Not Applicable
+
+Remediation: RKE2 doesn’t require or maintain a configuration file for etcd. All configuration is passed in as arguments at container run time.
+
+*1.1.8* Ensure that the etcd pod specification file ownership is set to `root:root` (Scored)
+
+Result: Not Applicable
+
+Remediation: RKE2 doesn’t require or maintain a configuration file for etcd. All configuration is passed in as arguments at container run time.
+
+*1.1.11* Ensure that the etcd data directory permissions are set to `700` or more restrictive (Scored)
+
+Result: Not Applicable
+
+Remediation: On the etcd server node, get the etcd data directory, passed as an argument `--data-dir`, from the below command:
+
+`ps -ef | grep etcd`
+
+Run the below command (based on the etcd data directory found above). For example,
+
+`chmod 700 /var/lib/etcd`
+
+Audit Script: `1.1.11.sh`
+
+```sh
+#!/bin/bash -e
+
+etcd_bin=${1}
+
+test_dir=$(ps -ef | grep ${etcd_bin} | grep -- --data-dir | sed 's%.*data-dir[= ]\([^ ]*\).*%\1%')
+
+docker inspect etcd | jq -r '.[].HostConfig.Binds[]' | grep "${test_dir}" | cut -d ":" -f 1 | xargs stat -c %a
+```
+
+Audit Execution:
+
+`./1.1.11.sh etcd`
+
+Expected result:
+
+    '700' is equal to '700'
+
+*1.1.12* Ensure that the etcd data directory ownership is set to `etcd:etcd` (Scored)
+
+Result: Not Applicable
+
+Remediation: On the etcd server node, get the etcd data directory, passed as an argument --data-dir, from the below command:
+
+`ps -ef | grep etcd`
+
+Run the below command (based on the etcd data directory found above). For example,
+
+`chown etcd:etcd /var/lib/etcd`
+
+Audit Script: `1.1.12.sh`
+
+```sh
+#!/bin/bash -e
+
+etcd_bin=${1}
+
+test_dir=$(ps -ef | grep ${etcd_bin} | grep -- --data-dir | sed 's%.*data-dir[= ]\([^ ]*\).*%\1%')
+
+docker inspect etcd | jq -r '.[].HostConfig.Binds[]' | grep "${test_dir}" | cut -d ":" -f 1 | xargs stat -c %U:%G
+```
+
+Audit Execution:
+
+`./1.1.12.sh etcd`
+
+Expected result:
+
+    'etcd:etcd' is present
+
+*1.1.13* Ensure that the admin.conf file permissions are set to 644 or more restrictive (Scored)
+
+Result: Not Applicable
```

Comment:
This is applicable. see https://github.com/rancher/security-scan/pull/22/files#r493592654